relay-server/src/metrics_extraction/spans/mod.rs

```diff
@@ -303,6 +312,28 @@ fn sanitized_span_description(
     Some(sanitized)
 }
 
+/// Trims the given span description to ensure
+/// [`relay-metrics::AggregatorService`] doesn't drop it for being too long.
+///
+/// If the description is short, it remains unchanged. If it's long, this method
+/// truncates it to the maximum allowed size and sets the last character to
+/// `*`.
+fn truncate_span_description(mut description: String, max_bytes: usize) -> String {
+    if bytecount::num_chars(description.as_bytes()) < max_bytes {
+        return description;
+    }
+
+    let mut cutoff = max_bytes - 1; // Leave space for `*`
+
+    while cutoff > 0 && description.is_char_boundary(cutoff) {
+        cutoff -= 1;
+    }
+
+    description.truncate(cutoff);
+    description.push('*');
+    description
+}
```

Comment:
Also, it would be nice to have a unit test that proves that special characters are truncated correctly (like @TBS1996 suggested, but I would target a specific case that would panic with a simple truncate).

Reply:
tried the fuzzy method and it fails :( 

```Rust
    #[test]
    fn test_no_panic_truncate() {
        let mut description = "".to_string();
        let chars = vec!['@', 'ي', 'Æ', '中', 'a'];

        for i in 0..200 {
            let ch = chars[i % chars.len()];
            description.push(ch);
        }

        for max_bytes in 10..100 {
            truncate_span_description(description.clone(), max_bytes);
        }
    }
   ```